server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -211,13 +211,14 @@ private boolean syncSegments() {
                 if (isRefreshAfterCommit()) {
                     remoteDirectory.deleteStaleSegmentsAsync(LAST_N_METADATA_FILES_TO_KEEP);
                 }
-
-                try (GatedCloseable<SegmentInfos> segmentInfosGatedCloseable = indexShard.getSegmentInfosSnapshot()) {
+                final Tuple<GatedCloseable<SegmentInfos>, ReplicationCheckpoint> tuple = indexShard.getLatestSegmentInfosAndCheckpoint();
+                try (GatedCloseable<SegmentInfos> segmentInfosGatedCloseable = tuple.v1()) {
+                    final ReplicationCheckpoint checkpoint = tuple.v2();
                     SegmentInfos segmentInfos = segmentInfosGatedCloseable.get();
-                    assert segmentInfos.getGeneration() == checkpoint.getSegmentsGen() : "SegmentInfos generation: "
-                        + segmentInfos.getGeneration()
-                        + " does not match metadata generation: "
-                        + checkpoint.getSegmentsGen();
+                    assert segmentInfos.getVersion() == checkpoint.getSegmentInfosVersion() : "SegmentInfos version: "
+                        + segmentInfos.getVersion()
+                        + " does not match metadata version: "
+                        + checkpoint.getSegmentInfosVersion();
```

Comment:
The version inside the ReplicationCheckpoint object seem to be created from the same segment infos that is being returned from the `getLatestSegmentInfosAndCheckpoint` method. Looks like the assertion evaluates to true (always?).

Reply:
you're right - this is leftover from when we were computing the checkpoint earlier.  Would like to take https://github.com/opensearch-project/OpenSearch/issues/9046 so we can simplify here.